tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleTooling.java

```diff
@@ -32,28 +32,42 @@
 /**
  * @author Gregory Amerson
  * @author Terry Jia
+ * @author Andy Wu
  */
 public class GradleTooling
 {
 
-    private static String getBundlePath( String bundleId )
+    private static String getBundlePath( String bundleId, File tempDir )
     {
         String installPath = Platform.getInstallLocation().getURL().getPath();
 
-        final String modelLocation = Platform.getBundle( bundleId ).getLocation();
+        final String bundleLocation = Platform.getBundle( bundleId ).getLocation();
 
-        String modelLocationString = modelLocation.replaceAll( "reference:", "" ).replaceAll( "file:", "" );
+        String bundleLocationString = bundleLocation.replaceAll( "reference:", "" ).replaceAll( "file:", "" );
 
-        File modelBundle = new File( modelLocationString );
+        File srcBundle = new File( bundleLocationString );
 
-        if( !modelLocationString.contains( installPath ) )
+        if( bundleLocationString.contains( "@" ) )
         {
-            modelBundle = new File( installPath, modelLocationString );
+            String p2Path = bundleLocationString.substring(
+                bundleLocationString.indexOf( "@" ) + 1, bundleLocationString.length() );
+            srcBundle = new File( p2Path );
+        }
+        else if( !bundleLocationString.contains( installPath ) )
+        {
+            srcBundle = new File( installPath, bundleLocationString );
+        }
+
+        File desBundle = new File( tempDir, srcBundle.getName() );
+
+        if( !desBundle.exists() )
+        {
+            FileUtil.copyFile( srcBundle, desBundle );
```

Comment:
Because the gradle daemon will hole the two jars inside eclipse plugin dir and after we close eclipse ,  the daemon is still running and we can't rename/delete the eclipse install dir. So I just copy the jars to temp dir and let the daemon hold them and will not affect eclipse itself.